lib/dialects/postgres/data-types.js

```diff
@@ -477,10 +477,11 @@ module.exports = BaseTypes => {
       let castKey = this.toSql();
 
       if (this.type instanceof BaseTypes.ENUM) {
-        castKey = `${Utils.addTicks(
-          Utils.generateEnumName(options.field.Model.getTableName(), options.field.fieldName),
-          '"'
-        ) }[]`;
+        const enumType = this.type.type || this.type;
```

Comment:
> * In `.../postgres/data-types.js` I wasn't a huge fan of the `type.type || type` stuff on line 480, but I just don't understand the code base enough to make an informed refactoring here.  Maybe it's fine, but just something I noticed as I was making changes.

Weird indeed, but are you sure this is necessary? What happens when using the code below instead?

```suggestion
        const enumType = this.type;
```

Reply:
Made the change, re-ran tests and still passed, so should be good!

Here are the other places where I've added this same check.  Tests failed when I removed either of these, so assume they are ok.

File: `/lib/dialects/postgres/query-generator.js`
Line 466: `const enumType = attribute.type.type || attribute.type;`
Line 764: `const type = dataType.type || dataType;`

Not sure if there is any refactoring that should be done here, but just wanted to make note that the reason I originally added these checks were because I saw the code below:

File: `/lib/dialects/postgres/query-interface.js`
Line 84: `const enumType = type.type || type;`